manage_rancher.py

```diff
@@ -262,20 +264,27 @@ def start_new(session: str, userid: str, prespawn: Optional[bool] = False):
     labels["traefik.enable"] = u"True"
     labels["session_id"] = session
     rule = u"Host(\"{}\") && PathPrefix(\"{}\") && HeadersRegexp(\"Cookie\",`{}`)"
-    labels["traefik.http.routers." + userid + ".rule"] = rule.format(cfg['hostname'], "/narrative", cookie)
+    labels["traefik.http.routers." + userid + ".rule"] = rule.format(cfg['hostname'], "/narrative/", cookie)
     labels["traefik.http.routers." + userid + ".entrypoints"] = u"web"
     container_config['launchConfig']['labels'] = labels
     container_config['launchConfig']['name'] = name
     container_config['launchConfig']['imageUuid'] = "docker:{}".format(cfg['image'])
     container_config['launchConfig']['environment'].update(cfg['narrenv'])
     container_config['name'] = name
     container_config['stackId'] = cfg['rancher_stack_id']
+
+    try:  # Set client ip from request object if available
+        request = flask.request
+        container_config['description'] = 'client-ip:{} timestamp:{}'.format(request.headers['X-Forwarded-For'],
+                                                                             datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%fZ'))
+    except Exception:
```

Comment:
Usually something should be done with an exception instead of just pass. Maybe set container_config['description'] to be unknown Or empty? 
I'm not clear on what can cause the exception here.

Reply:
container_config['description'] defaults to null in the code above this. But I've added a log message flagging the problem. The only way I could imagine this triggering an exception would be if this was called outside of a flask request context (for example, during service startup).

Reply:
It is not explicitly set to null, it is set to undefined. To me it depends on if it's a required value or not for the dictionary.

See the following code snippet
```
config = {}

try:
  config['required_description'] = undefined_variable
except:
   pass

description2 = config['required_description']
```
would raise
`KeyError: 'required_description'` unless you always did .get()

Reply:
Its not required - ithe rancher API has been happily taking that request, and it is "null" by default when you query the API.